agglayer/types.go

```diff
@@ -511,6 +643,35 @@ func (c *ClaimFromRollup) String() string {
 		c.ProofLeafLER.String(), c.ProofLERToRER.String(), c.ProofGERToL1Root.String(), c.L1Leaf.String())
 }
 
+type ClaimSelector struct {
+	obj Claim
+}
+
+func (c *ClaimSelector) GetObject() Claim {
+	return c.obj
+}
+
+func (c *ClaimSelector) UnmarshalJSON(data []byte) error {
+	var obj map[string]interface{}
+	if string(data) == nullStr {
+		return nil
+	}
+	err := json.Unmarshal(data, &obj)
+	if err != nil {
```

Comment:
I've seen couple of places where you can do this, just to save couple of lines of code:

```suggestion
	if err := json.Unmarshal(data, &obj); err != nil {
```

Reply:
I changed  a lot of them